testgrid/transfigure.sh

```diff
@@ -0,0 +1,98 @@
+#!/bin/bash
+
+# A script that uses configurator to create a YAML file, and then pushes that
+# YAML file to k8s/test-infra's config/testgrids directory
+#
+# Used by teams that host different instances of Prow, but want to show their
+# results on testgrid.k8s.io
+#
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+main() {
+  branch="transfigure-branch"
+
+  if [[ $# -lt 1 ]]; then
+      echo "Usage: $(basename "$0") </path/to/github/token> [prow_config] [prow-job_config] [testgrid-yaml] [subdir-path]"
+      echo "Path to GitHub token required"
+      exit 1
+  fi
+
+  token=$(readlink -f $1)
+  shift
+  user-from-token
+  ensure-git-config
+  parse-args "$@"
+
+  echo "Getting kubernetes/test-infra repo"
+  git clone https://github.com/kubernetes/test-infra.git || echo "Found test-infra directory; proceeding"
+  k8srepo=$(readlink -f test-infra)
+  target=$(readlink -f test-infra/config/testgrids)
+  trap "cleanup-repository" ERR
```

Comment:
Change this to exit. This will allow `cleanup-repository` to run on script exit (which includes error since `errexit` is enabled). Now code paths like [line 53](https://github.com/kubernetes/test-infra/pull/14967/files#diff-47de6d2ae75ed815b50c5854ae32cf2dR53) will be cleaned up properly. Also, you can get rid of [line 71](https://github.com/kubernetes/test-infra/pull/14967/files#diff-47de6d2ae75ed815b50c5854ae32cf2dR71) afterward.
```suggestion
  trap "cleanup-repository" EXIT
```